src/applications/disability-benefits/526EZ/components/AsyncDisplayWidget.jsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+
+import LoadingIndicator from '@department-of-veterans-affairs/formation/LoadingIndicator';
+import AlertBox from '@department-of-veterans-affairs/formation/AlertBox';
+
+const PENDING = 'PENDING';
+const RESOLVED = 'RESOLVED';
+const REJECTED = 'REJECTED';
+
+
+/**
+ * Handles the various display statuses when calling an asynchronous function.
+ */
+export default class AsyncDisplayWidget extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: null,
+      promiseState: PENDING
+    };
+
+    // Make sure we configured it correctly
+    // StringField passes 'ui:options' as the options prop
+    const uiOptions = props.options;
+    if (!uiOptions) {
+      throw new Error('No ui:options supplied to AsyncDisplayWidget.');
+    }
+
+    if (typeof uiOptions.viewComponent !== 'function') {
+      throw new Error('AsyncDisplayWidget requires viewComponent in ui:options to be a React component.');
+    }
+
+    if (typeof uiOptions.callback !== 'function') {
+      throw new Error('AsyncDisplayWidget requires callback in ui:options to be a function.');
+    }
+
+    // TODO: Don't call the callback _every_ time the component is mounted
+    const cbPromise = uiOptions.callback();
```

Comment:
This is a tough one to name, can we change the name of `cbPromise` to something that describes what it does rather than what it is?

Reply:
Heh, I'm wide open to suggestions! I couldn't think of anything better, though. :disappointed:

Reply:
We seem to use `apiRequest` pretty frequently as a name for a similar thing. Perhaps `requestHandler` or some such? @jbalboni any other ideas?